### ISSUE TITLE
"Variable is assigned to, but never used" with += operator

### ISSUE THREAD

**Issue by ColinPitrat** [rust-lang/rust#63046]
Tested on both rust 1.36 and 1.38 (nightly):
$ rustc -V
rustc 1.36.0 (a53f9df32 2019-07-03)
$ rustc +nightly -V
rustc 1.38.0-nightly (c43753f91 2019-07-26)

The following code:

```
$ cat src/main.rs 
fn main() {
    let mut error = 0;
    for i in 0..3 {
        error += i;
    }
}
```

Produces the following warning:

```
warning: variable `error` is assigned to, but never used
 --> src/main.rs:2:13
  |
2 |     let mut error = 0;
  |             ^^^^^
  |
  = note: #[warn(unused_variables)] on by default
  = note: consider using `_error` instead
```

Which is incorrect. The value assigned, 0.0 is obviously used as the code is effectively the following (which doesn't produce a warning):

```
error = error + i;
```

**Comment by BO41** [comment]
I am not sure if I understand you correctly. But this warning _is_ correct.
You are assigning to `error`, but you never actually do anything with the value.
> warning: variable `error` is assigned to, but never used
>  --> src/main.rs:2:13
>   |
> 2 |     let mut error = 0;
>   |             ^^^^^
>   |
>   = note: #[warn(unused_variables)] on by default
>   = note: consider using `_error` instead

This should throw the same error, but it doesn't
```
fn main() {
    let mut error = 0;
    for i in 0..3 {
        error = error + i;
    }
}
```
That's the actual bug

**Comment by ColinPitrat** [comment]
Indeed, I confused the warning for another, the following doesn't raise the warning:

```
$ cat  src/main.rs
fn main() {
    let mut error = 0;
    for i in 0..3 {
        error += i;
    }
    println!("{}", error);
}
```

I guess detecting the second one is harder. I'll close this issue, I'll let you reopen if you think the "actual bug" should be fixed.

**Comment by BO41** [comment]
> I guess detecting the second one is harder.

What do you mean with that?

```rust
fn main() {
    let mut error = 0;
    for i in 0..3 {
        error += i;
    }
    println!("{}", error);
}
```
you can't omit the `= 0;`, if you mean this.
